components/todoist/sources/common-task.mjs

```diff
@@ -0,0 +1,38 @@
+import common from "./common.mjs";
+
+export default {
+  ...common,
+  props: {
+    ...common.props,
+    selectProjects: {
+      propDefinition: [
+        common.props.todoist,
+        "selectProjects",
+      ],
+    },
+  },
+  methods: {
+    ...common.methods,
+    isElementRelevant() {
+      return true;
+    },
+    async getSyncResult() {
+      return this.todoist.syncItems(this.db);
+    },
+    filterResults(syncResult) {
+      return Object.values(syncResult)
+        .filter(Array.isArray)
+        .flat()
+        .filter(this.isElementRelevant)
+        .filter((element) =>
+          this.todoist.isProjectInList(element.project_id, this.selectProjects));
```

Comment:
When I deploy or run the [`new-or-modified-task`](https://github.com/PipedreamHQ/pipedream/blob/daa0a3424a0a1352fa21704b6c4aefdf043eff2e/components/todoist/sources/new-or-modified-task/new-or-modified-task.mjs) source, and leave the `selectProjects` prop blank, I get the error
```
TypeError: Cannot read property 'length' of undefined
```
<img width="700" alt="todoist-typeerror2" src="https://user-images.githubusercontent.com/19861096/151046724-a9db8e47-3955-4874-9d20-b7f10003541f.png">

at this line:
https://github.com/PipedreamHQ/pipedream/blob/daa0a3424a0a1352fa21704b6c4aefdf043eff2e/components/todoist/todoist.app.mjs#L330

To fix, I'd suggest:
```suggestion
          this.todoist.isProjectInList(element.project_id, this.selectProjects ?? []));
```